src/ripple/consensus/Validations.h

```diff
@@ -590,9 +606,45 @@ class Validations
             // validations sequence from that validator
             auto const now = byLedger_.clock().now();
             SeqEnforcer<Seq>& enforcer = seqEnforcers_[nodeID];
+
             if (!enforcer(now, val.seq(), parms_))
                 return ValStatus::badSeq;
 
+            {
+                auto& vs = bySequence_[val.seq()];
+
+                auto const [it, inserted] = vs.emplace(nodeID, val);
+
+                if (!inserted)
+                {
+                    // Check if it's expired; if it is, replace it. We need to
+                    // calculate the difference between the sign times for the
+                    // two validations:
+                    auto const diff =
+                        std::max(it->second.signTime(), val.signTime()) -
+                        std::min(it->second.signTime(), val.signTime());
+
+                    if (diff > std::chrono::minutes{10} &&
```

Comment:
Consider using one of the other parameterized durations over this hardcoded 10 minutes.